processing/src/main/java/org/apache/carbondata/processing/newflow/iterators/RecordReaderIterator.java

```diff
@@ -0,0 +1,40 @@
+package org.apache.carbondata.processing.newflow.iterators;
+
+import java.io.IOException;
+
+import org.apache.carbondata.common.CarbonIterator;
+import org.apache.carbondata.common.logging.LogService;
+import org.apache.carbondata.common.logging.LogServiceFactory;
+
+import org.apache.hadoop.mapred.RecordReader;
+
+/**
+ * This iterator iterates RecordReader.
+ */
+public class RecordReaderIterator extends CarbonIterator<Object[]> {
```

Comment:
I thought this PR is for loading part, why RecordReader related abstraction is required?